src/test/java/com/google/googleidentity/oauth2/client/InMemoryClientDetailsServiceTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.client;
+
+import com.google.common.base.Charsets;
+import com.google.common.hash.Hashing;
+import com.google.googleidentity.oauth2.client.ClientDetails;
+import com.google.googleidentity.oauth2.client.ClientDetailsService;
+import com.google.googleidentity.oauth2.client.InMemoryClientDetailsService;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test {@link InMemoryClientDetailsService}
+ */
+public class InMemoryClientDetailsServiceTest {
+
+    private static final ClientDetails client =
```

Comment:
Should be upper case https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names
```suggestion
    private static final ClientDetails CLIENT =
```